src/Security/OidcAuthenticator.php

```diff
@@ -63,8 +64,18 @@ public function authenticate(Request $request): Passport
       // Retrieve the user data with the authentication data
       $userData = $this->oidcClient->retrieveUserInfo($authData);
 
+      // Look for the user identifier in either the id_token or the userinfo endpoint
+      if ($this->userIdentifierFromIdToken) {
+        $tokenData = $this->oidcClient->retrieveDataFromIdToken($authData);
+        $userIdentifier = array_key_exists($this->userIdentifierProperty, $tokenData)
+            ? $tokenData[$this->userIdentifierProperty]
+            : null;
+      } else {
+        $userIdentifier = $userData->getUserDataString($this->userIdentifierProperty);
+      }
```

Comment:
With this code I would prefer the tertiary assignment.

```php
      $userIdentifier = $this->userIdentifierFromIdToken
        ? $this->oidcClient->retrieveDataFromIdToken($authData)[$this->userIdentifierProperty] ?? null
        : $userData->getUserDataString($this->userIdentifierProperty);
```

Reply:
But (combined with the other comments actually), lets use the jwt helper directly here, something like: 

```php
$this->jwtHelper->getIdTokenClaims($authData)->{$this->userIdentifierProperty} ?? null;
```

The JwtHelper will need to be injected in the authenticator as well, but it can be an optional dependency with null default. That way we do not break BC, but we should throw when it is null and the `userIdentifierFromIdToken` is set to `true`. So then it becomes something like this:

```php
if ($this->userIdentifierFromIdToken) {
  if (!$this->jwtHelper) {
    throw new RuntimeException('Missing required JwtHelper object when retrieving user identifier from token. Did you extend the authenticator and not provide the JwtHelper?')
  }

  $userIdentifier = $this->jwtHelper->getIdTokenClaims($authData)->{$this->userIdentifierProperty} ?? null;
} else {
  $userIdentifier = $userData->getUserDataString($this->userIdentifierProperty);
}
```